pitop/miniscreen/oled/core/canvas.py

```diff
@@ -49,7 +49,7 @@ def image(self, xy, image):
         :return: The current canvas pixel map as a 2D array
         :rtype: array
         """
-        image_data = process_pil_image(image)
+        image_data = process_pil_image(image, size=self.get_size(), mode="1")
```

Comment:
This is setting the mode to 1-bit explicitly here - would probably be better to get it from `self.__oled_device.mode`.
Also, this file has `Image helper methods` which should move into the `ImageHelper` class and create a more comprehensive and useful library of functions.